packages/storybook/src/utils/testing.ts

```diff
@@ -92,3 +107,110 @@ export class TestButtonComponent implements OnInit {
   );
   return appTree;
 }
+
+function getTempDir() {
+  const tmpDir = tmpdir();
+  const tmpFolder = `${tmpDir}${sep}`;
+  return mkdtempSync(tmpFolder);
+}
+
+export async function getTestArchitect() {
+  const tmpDir = getTempDir();
+  const architectHost = new TestingArchitectHost(tmpDir, tmpDir);
+  const registry = new schema.CoreSchemaRegistry();
+  registry.addPostTransform(schema.transforms.addUndefinedDefaults);
+
+  const architect = new Architect(architectHost, registry);
+
+  await architectHost.addBuilderFromPackage(join(__dirname, '../..'));
+
+  return [architect, architectHost] as [Architect, TestingArchitectHost];
+}
+
+export async function getMockContext() {
+  const [architect, architectHost] = await getTestArchitect();
+
+  const context = new MockBuilderContext(architect, architectHost);
+  await context.addBuilderFromPackage(join(__dirname, '../..'));
+  return context;
+}
+
+export class MockBuilderContext implements BuilderContext {
+  id: 0;
+
+  builder: any = {};
+  analytics = null;
+
+  target: Target = {
+    project: null,
+    target: null
+  };
+
+  get currentDirectory() {
+    return this.architectHost.currentDirectory;
+  }
+  get workspaceRoot() {
+    return this.architectHost.workspaceRoot;
+  }
+  logger = new TestLogger('test');
+  constructor(
+    private architect: Architect,
+    private architectHost: TestingArchitectHost
+  ) {}
+
+  async addBuilderFromPackage(path: string) {
+    return await this.architectHost.addBuilderFromPackage(path);
+  }
+
+  async addTarget(target: Target, builderName: string) {
+    return await this.architectHost.addTarget(target, builderName);
+  }
+
+  getBuilderNameForTarget(target: Target) {
+    return this.architectHost.getBuilderNameForTarget(target);
+  }
+
+  scheduleTarget(
+    target: Target,
+    overrides?: JsonObject,
+    scheduleOptions?: ScheduleOptions
+  ) {
+    return this.architect.scheduleTarget(target, overrides, scheduleOptions);
+  }
+
+  scheduleBuilder(
+    name: string,
+    overrides?: JsonObject,
+    scheduleOptions?: ScheduleOptions
+  ) {
+    return this.architect.scheduleBuilder(name, overrides, scheduleOptions);
+  }
+
+  getTargetOptions(target: Target) {
+    return this.architectHost.getOptionsForTarget(target);
+  }
+
+  validateOptions<T extends JsonObject = JsonObject>(
+    options: JsonObject,
+    builderName: string
+  ): Promise<T> {
+    return Promise.resolve<T>(options as T);
+  }
+
+  reportRunning() {}
+
+  reportStatus(status: string) {}
+
+  reportProgress(current: number, total?: number, status?: string) {}
+
+  addTeardown(teardown: () => Promise<void> | void) {}
+
+  async getProjectMetadata(
```

Comment:
Sorry for the inconvenience:

Can you please merge in the latest master and remove this method.

Your branch has changes (angular-devkit 9.0.0) which has been reverted for now.

This method is not available yet. I can handle adding it back when it is.